### PR TITLE
[UXIT-1319] CMS - Fix Featured Entries

### PR DESCRIPTION
**Pull request by mirhamasala**
This PR addresses the handling of featured entries by ensuring that we correctly extract the slug from the provided file path. The featured entries field now uses the full file path, allowing us to dynamically parse and obtain the slug for further operations, ensuring compatibility with file-based CMS structures.

Screenshot 2024-08-07 at 10 48 27

Screenshot 2024-08-07 at 10 48 52

Screenshot 2024-08-07 at 10 48 11

Screenshot 2024-08-07 at 10 49 15

### CODE REVIEW

File: public/admin/config.yml [hunk 1]
```diff
@@ -140,8 +140,8 @@ collections:
           - name: "featured_entry"
             label: "Featured Post"
             widget: "relation"
-            collection: "blog"
-            value_field: "slug"
+            collection: "blog_posts"
```
Comment (CharlyMartin) [inline public/admin/config.yml:143]: We use both `_` and `-` for spaces. I think we should harmonize that.

File: src/app/_types/cmsConfig.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-export type CMSCollectionName = 'blog' | 'ecosystem' | 'events'
+export type CMSCollectionName = 'blog_posts' | 'ecosystem' | 'events'
```
Comment (CharlyMartin) [inline src/app/_types/cmsConfig.ts]: Shouldn't `events` be `event-entries` now if we follow the same logic?

Also, I think `events` is a self-explanatory name.

Comment (mirhamasala) [reply src/app/_types/cmsConfig.ts]: Yes, it should be `event_entries`. The problem with just using `events` is that we also have the Events page, which is also called `events` in the CMS config.

File: public/admin/config.yml (hunk 1 above)
Comment (mirhamasala) [reply public/admin/config.yml:143]: I was trying to use `_` in the CMS config for fields related to names - This way, it's very easy to spot in our codebase that this is CMS data related. We can definitely look into harmonizing.

File: src/app/_types/cmsConfig.ts (hunk 2 above)
Comment (mirhamasala) [reply src/app/_types/cmsConfig.ts]: Actually, on my side it's already `event_entries` - Probably a CR that didn't get there in time.

**Comment by CharlyMartin** [comment]
@mirhamasala I don't fully understand why these changes are needed but it works on my side!

**Comment by mirhamasala** [comment]
> @mirhamasala I don't fully understand why these changes are needed but it works on my side!

At the moment, if you try to change the featured entries for the homepage, Blog page, Events page, or Grants page, in the CMS - It won't work.

**Review by CharlyMartin — APPROVED** [review]

**Pull request merged**